src/popup/components/ad-hoc-tools-panel.tsx

```diff
@@ -1,75 +1,64 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-import { Icon } from 'office-ui-fabric-react';
-import { Link } from 'office-ui-fabric-react';
+import { NamedFC } from 'common/react/named-fc';
+import { flatMap } from 'lodash';
+import { css, Icon, Link } from 'office-ui-fabric-react';
 import * as React from 'react';
-
+import * as styles from './ad-hoc-tools-panel.scss';
 import { DiagnosticViewToggleFactory } from './diagnostic-view-toggle-factory';
 
 export interface AdHocToolsPanelProps {
     backLinkHandler: () => void;
     diagnosticViewToggleFactory: DiagnosticViewToggleFactory;
 }
 
-export class AdHocToolsPanel extends React.Component<AdHocToolsPanelProps, {}> {
-    private sliceSize: number = 3;
+export const AdHocToolsPanel = NamedFC<AdHocToolsPanelProps>('AdHocToolsPanel', props => {
+    const getTogglesWithDividers = () => {
```

Comment:
A nice further improvement would be to move these dividers to be CSS borders rather than elements in the layout, but I think it's reasonable to defer that for a separate PR

Reply:
Agree... I'll have a second look as soon as I have some slack time.